xarray/core/indexing.py

```diff
@@ -1076,7 +1076,13 @@ def _logical_any(args):
     return functools.reduce(operator.or_, args)
 
 
-def _masked_result_drop_slice(key, chunks_hint=None):
+def _masked_result_drop_slice(key, data=None):
+
+    if data is not None:
+        chunks_hint = getattr(data, "chunks", None)
+    else:
+        chunks_hint = None
```

Comment:
The getattr line above already does this. I don't think you need this if statement at all.